### PR TITLE
Enhancement/ab2 d 3720 testing patient endpoint and r4

### PR DESCRIPTION
**Pull request by sharshar-sb**
**JIRA Tickets:**

[AB2D-3720](https://jira.cms.gov/browse/AB2D-3720) - Tested STU3 & R4
 
### What Does This PR Do?

Adds testing for jobs for both STU3 & R4. Also added testing of patient endpoints although they are not running since they would fail. Also added year parameter patient search.

### Security Implications

- [ ] This PR adds new software dependencies
- [ ] This PR modifies or invalidates our security controls
- [ ] This PR stores or transmits data that was not stored or transmitted before

- [ ] This PR requires additional review of its security implications for other reasons

### Submitter Checklist

- [ ] This PR includes any required documentation changes, (`README`, changelog / release notes, website).
- [ ] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments.
- [ ] Code checked for PHI/PII exposure

### CODE REVIEW

**Review by nflearl — APPROVED** [review]

**Review by nflearl — APPROVED** [review]

File: e2e-test/src/test/java/gov/cms/ab2d/e2etest/TestRunner.java [hunk 1]
```diff
@@ -803,7 +804,7 @@ void testClientCannotMakeRequestWithNullClaims(FhirVersion version) throws IOExc
     @MethodSource("getVersion")
     @Order(14)
     void testBadQueryParameterResource(FhirVersion version) throws IOException, InterruptedException {
-        System.out.println("Starting test 14");
+        System.out.println("Starting test 14 - " + version.toString());
```
Comment (sb-wnyffenegger) [inline e2e-test/src/test/java/gov/cms/ab2d/e2etest/TestRunner.java:807]: Any way to use lombok to get slf4j logging for this stuff?

File: e2e-patient-test/src/test/java/gov/cms/ab2d/SpringBootApp.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package gov.cms.ab2d;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class SpringBootApp {
```
Comment (sb-wnyffenegger) [inline e2e-patient-test/src/test/java/gov/cms/ab2d/SpringBootApp.java:7]: Do we really need this to run the e2 tests?

File: e2e-patient-test/pom.xml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
```
Comment (sb-wnyffenegger) [inline e2e-patient-test/pom.xml:5]: Larger question does this need to be a separate JAR from the other e2e tests?

File: e2e-patient-test/src/test/java/gov/cms/ab2d/SpringBootApp.java (hunk 2 above)
Comment (sharshar-sb) [reply e2e-patient-test/src/test/java/gov/cms/ab2d/SpringBootApp.java:7]: So, it's not technically e2e as much as it requires outside resources (BFD sandbox) to run. It auto-wires the BFD client. I wanted it separate from the bfd module.

File: e2e-patient-test/pom.xml (hunk 3 above)
Comment (sharshar-sb) [reply e2e-patient-test/pom.xml:5]: Trying to use the existing e2e tests and have the correct configuration for the SpringBootTest that the patient endpoint tests needs was challenging and I didn't want to mess too much with the e2e tests.

**Review by sb-wnyffenegger — APPROVED** [review]

**Pull request merged**